lib/errors/AuthApiError.js

```diff
@@ -10,19 +10,21 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-function AuthApiError(err, xhr) {
-  this.name = 'AuthApiError';
-  this.message = err.errorSummary;
-  this.errorSummary = err.errorSummary;
-  this.errorCode = err.errorCode;
-  this.errorLink = err.errorLink;
-  this.errorId = err.errorId;
-  this.errorCauses = err.errorCauses;
+export default class AuthApiError extends Error {
+  constructor(err, xhr = null) {
```

Comment:
is the default value needed?

Reply:
We could use a named export. It would just need to be adjusted. Do you think in general we should not use `default` exports?

Reply:
I mean default value for `xhr`

Reply:
Oh, yes it could be `xhr?`, I will just need to change extension to `.ts`. I think I will probably change all the `.js` files under `/lib` to `.ts` so that we don't have to deal with that nuance.